orc8r/cloud/go/services/state/servicers/servicer.go

```diff
@@ -225,7 +237,28 @@ func (srv *stateServicer) getStates(_ context.Context, req *protos.GetStatesRequ
 	return &protos.GetStatesResponse{States: blobsToStates(blobs)}, nil
 }
 
-func (srv *stateServicer) searchStates(_ context.Context, req *protos.GetStatesRequest) (*protos.GetStatesResponse, error) {
+func (srv *cloudStateServicer) getStates(_ context.Context, req *protos.GetStatesRequest) (*protos.GetStatesResponse, error) {
+	store, err := srv.factory.StartTransaction(nil)
+	if err != nil {
+		return nil, internalErr(err, "GetStates (get) blobstore start transaction")
+	}
+
+	ids := idsToTKs(req.GetIds())
+	blobs, err := store.GetMany(req.GetNetworkID(), ids)
+	if err != nil {
+		_ = store.Rollback()
+		return nil, internalErr(err, "GetStates (get) blobstore get many")
+	}
+
+	err = store.Commit()
+	if err != nil {
+		return nil, internalErr(err, "GetStates (get) blobstore commit transaction")
+	}
+
+	return &protos.GetStatesResponse{States: blobsToStates(blobs)}, nil
+}
```

Comment:
Why adding this in? Unless I'm reading wrong, looks unusued

Reply:
used on line 69 of this file

Reply:
Ah gotcha. I guess I'm just confused why GitHub thinks this block is new? But whatever, seems reasonable given my skimming of the original file 👍

Reply:
Oh I just copied the method for both cloudStateServicer & stateServicer. Don't think there is a good way to share the impl of methods on structs, like I can't import one to another...

Reply:
Ah gotcha. Okay yeah is fine